modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/writer/BatchEngineTaskItemWriterFactory.java

```diff
@@ -175,6 +175,8 @@ public void removedService(
 			ServiceReference<Object> serviceReference, List<String> keys) {
 
 			keys.forEach(_resourceServiceReferenceMap::remove);
+
+			_bundleContext.ungetService(serviceReference);
```

Comment:
@shuyangzhou why do we need to unget a service as it is not created in addService method? All instances created via ServiceObjects are unget after usage in BatchEngineTaskItemWriter.close mathod.

Reply:
@ivicac There is a get service call in the addingService https://github.com/brianchandotcom/liferay-portal/pull/78512/files#diff-c588efcdcc62056884b280778f566da5R116